engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/BuiltinRootNode.java

```diff
@@ -12,6 +13,29 @@ protected BuiltinRootNode(EnsoLanguage language) {
     super(language);
   }
 
+  protected QualifiedName moduleName = null;
+  protected QualifiedName typeName = null;
+
+  /** Get the module name where the builtin is defined. */
+  public QualifiedName getModuleName() {
+    return moduleName;
+  }
+
+  /** Set the module name where the builtin is defined. */
+  public void setModuleName(QualifiedName moduleName) {
+    this.moduleName = moduleName;
```

Comment:
I don't like mutability of things that should be immutable. I'd prefer to remove the setters completely. If that is not possible, please throw exception if the value is already set and somebody wants to redefine it.

Reply:
Ok, I'll update the setters in the next PR